### DISCUSSION TITLE
Laravel MongoDB Search Problem

### DISCUSSION THREAD

**Discussion by niyazialpay** [meilisearch/product#708] [Feedback & Feature Proposal]
Hello,

I use my Laravel project with MongoDB database. The project works correctly without any incompatibility and problems with MongoDB on the Laravel side, but there is a problem with Meilisearch integration. Added or updated content reaches the Meilisearch side. However, Laravel cannot bring the content because the query to the database is made incorrectly in the search process with the model.

use App\Models\Order;
$orders = Order::search('Star Trek')->get();

When I examine this with Laravel debugbar, I see that this query result is executed on the MongoDB side as follows.

order.find({"order._id":{"$in":["656d92be76e5080bcf02c393"]}},{"typeMap":{"root":"array","document":"array"}})

On the MongoDB side, only field should be used in the find query, when the query is queried as order._id, it works as a nested query and no result comes. It would be very good if you can make an update here to query only with field without table name.

**Comment by macraig** [comment]
Hi @niyazialpay , apologies for the late reply.  It sounds like this might be a feature request for the https://github.com/laravel/scout repository?

**Reply by niyazialpay** [reply]
Thank you for your reply. Actually I fixed this problem with create a new trait. I copy the '/vendor/laravel/scout/src/Searchable.php' file to the app directory and changed below lines

 '$self->qualifyColumn($self->getScoutKeyName())' to '$self->getScoutKeyName()',

'$this->qualifyColumn($this->getScoutKeyName()), $ids' to '$this->getScoutKeyName(), $ids'.

After all this I wrote a blog about my solution

[https://niyazi.net/en/laravel-meilisearch-and-mongodb-integration](https://niyazi.net/en/laravel-meilisearch-and-mongodb-integration)

I created both pull request and feature request.

**Reply by macraig** [reply]
Amazing news! Thanks for sharing it with us!

**Comment by GromNaN** [comment]
Laravel Scout works with MongoDB models using the `mongodb/laravel-mongodb` package. The method [`qualifyColumn()`](https://github.com/mongodb/laravel-mongodb/blob/9973195cd44e85d7684638d631b6bc148decaf34/src/Eloquent/DocumentModel.php#L563-L567) is customized to not alter the column name.
